### ISSUE TITLE
Unexpected termination when using HSE functional

### ISSUE THREAD

**Issue by ErjieWu** [deepmodeling/abacus-develop#4983]
### Describe the bug

When I run ABACUS with `dft_functional = hse` and `out_mat_hs2 = 1`, the programme will be killed after leaving scf.
![image](https://github.com/user-attachments/assets/9920590f-8b7e-4819-a6c9-94ef2ee3f3ed)
If `out_mat_hs2` is not set to 1, the program ends normally.

For compiling, I used toolchain and following the step in https://mcresearch.github.io/abacus-user-guide/abacus-oneapi.html

### To Reproduce

INPUT:
```
INPUT_PARAMETERS
#Parameters (1.General)
suffix          Si-fcc-scf
calculation     scf
esolver_type    ksdft
pseudo_dir      ./
orbital_dir     ./
dft_functional  hse
cal_stress      1
cal_force       1
stru_file       PRIMCELL.STRU
#Parameters (2.Iteration)
ecutwfc         100
scf_thr         1e-8
scf_nmax        100
#Parameters (3.Basis)
basis_type      lcao
gamma_only      0
#Parameters (4.Smearing)
smearing_method mp
smearing_sigma  0.015
#Parameters (5.Mixing)
mixing_type     broyden
mixing_beta     0.7

out_chg         1
out_mat_hs2     1
out_mat_r       1
```

PRIMCELL.STRU
```
ATOMIC_SPECIES
Si   28.085  Si_ONCV_PBE-1.0.upf

NUMERICAL_ORBITAL
Si_gga_7au_100Ry_2s2p1d.orb

LATTICE_CONSTANT
1.889726

LATTICE_VECTORS
0.000000000000  2.720626132599  2.720626132599
2.720626132599  0.000000000000  2.720626132599
2.720626132599  2.720626132599  0.000000000000

ATOMIC_POSITIONS
Direct

Si
0.000
2
0.000000000000  0.000000000000  0.000000000000  1  1  1  mag  0.0
0.250000000000  0.250000000000  0.250000000000  1  1  1  mag  0.0
```

KPT
```
K_POINTS
0
Gamma
8 8 8 0 0 0
```

### Task list for Issue attackers (only for developers)

- [ ] Verify the issue is not a duplicate.
- [ ] Describe the bug.
- [ ] Steps to reproduce.
- [ ] Expected behavior.
- [ ] Error message.
- [ ] Environment details.
- [ ] Additional context.
- [ ] Assign a priority level (low, medium, high, urgent).
- [ ] Assign the issue to a team member.
- [ ] Label the issue with relevant tags.
- [ ] Identify possible related issues.
- [ ] Create a unit test or automated test to reproduce the bug (if applicable).
- [ ] Fix the bug.
- [ ] Test the fix.
- [ ] Update documentation (if necessary).
- [ ] Close the issue and inform the reporter (if applicable).

**Comment by QuantumMisaka** [comment]
@ErjieWu What are your ABACUS and LibRI version ?

**Comment by maki49** [comment]
This have been fixed by #4956.
